### PR TITLE
AGP legal instrument list view 

### LINKED ISSUE #906: As an AGP user I want to explore secondary legal materials
These are non-primary legal materials, such as reports, strategy documents, books and guides.

* [x] new top-level navigation: "Reports & Guides" at /doc to replace the existing "AU Documents"
* [ ] change page title to "Reports and Guides"
* [ ] use the standard GenericDocumentView, but show only `frbr_uri_doctype="doc"`
* [ ] use existing "regional body" and "nature" facets

See https://docs.google.com/spreadsheets/d/1s5t_5dMFCblO7DP8elqsJTR9K84IvbjE/edit#gid=1645975557 for full doctype breakdown and context

### LINKED ISSUE #907: As an AGP user I want to explore Soft Law information
Soft law includes: hansard, decisions, model laws, recommendations.

* [ ] a new top-level navigation for "Soft Law" at /soft-law to replace "Treaties and Protocols"
* [ ] use the existing GenericDocumentView and *exclude* `frbr_uri_doctype="doc"`
* [ ] same faceting as existing GenericDocumentView

This is the complement of #906 

See https://docs.google.com/spreadsheets/d/1s5t_5dMFCblO7DP8elqsJTR9K84IvbjE/edit#gid=1645975557 for full doctypes and context.

### LINKED ISSUE #908: AGP legislation view updates
`/legal_instruments` is to be changed to `/legal-instruments` and cover both Legislation and LegalInstrument document types. In other words, this shows all documents with the `act`  FRBR URI doc type.

* [x] new top-level nav "Legal Instruments" under "African Legal Information" to replace "Legislation" (already done)
* [ ] change the page's model to be `CoreDocument` and filter on `frbr_uri_doctype="act"`.
* [ ] facet on nature and show nature in document table (similar to /doc)

### PR DESCRIPTION
**Pull request by actlikewill**
### Legal Instrument Listing
- [x] Change legal instrument list view to include legislation
- [x] Filter on `frbr_uri_doctype="act"`
- [x] Change header link

image

### Soft Law Listing
- [x] Add soft-law view to agp
- [x] Exclude `frbr_uri_doctype="doc"` on softlaw
- [x] Add listing template
- [x] Change  link in header

image

### Reports and Guides

- [x] Add Reports and Guides view to agp
- [x] Filter on `frbr_uri_doctype="doc"`
- [x] Add listing template
- [x] Change link in header

image

closes #906 
closes #907
closes #908

### CODE REVIEW

**Comment by actlikewill** [comment]
@longhotsummer Does the title of the page need to change? Right now it reads Treaties and Protocols. Also I have not changed the legislation/ view it still lists the legislation model. I can see that there is a link to it labeled 'Soft Law'.

**Comment by longhotsummer** [comment]
Fixes #908

File: africanlii/views.py [hunk 1]
```diff
@@ -16,6 +17,21 @@ def get_context_data(self, **kwargs):
         )
 
 
+class AGPLegalInstrumentListView(FilteredDocumentListView):
+    model = CoreDocument
+    template_name = "peachjam/legal_instrument_list.html"
+
+    def get_queryset(self):
```
Comment (longhotsummer) [inline africanlii/views.py:24]: I think this must be get_base_queryset()

File: africanlii/urls.py [hunk 2]
```diff
@@ -7,5 +7,10 @@
     path(
         "legislation/", views.AGPLegislationListView.as_view(), name="legislation_list"
```
Comment (longhotsummer) [inline africanlii/urls.py:8]: This view falls away

**Comment by longhotsummer** [comment]
> @longhotsummer Does the title of the page need to change? Right now it reads Treaties and Protocols. 

Yes, please update the title to "Legal Instruments" to match the nav.

> Also I have not changed the legislation/ view it still lists the legislation model. I can see that there is a link to it labeled 'Soft Law'.

Apologies, this issue was written before the main nav was updated, and it's a bit confusing. This view replaces the existing "/legal_instruments" view. It combines both legislation and legal instruments into one listing, with faceting by doc nature (subtype) -- see [here](https://docs.google.com/spreadsheets/d/1s5t_5dMFCblO7DP8elqsJTR9K84IvbjE/edit#gid=1645975557)

I've updated the original issue description.

The conversion of "/legislation" into "/soft-law" will be handled by https://github.com/laws-africa/peachjam/issues/907

File: africanlii/urls.py [hunk 3]
```diff
@@ -7,5 +7,10 @@
...
     ),
+    path(
+        "legal_instruments/",
```
Comment (longhotsummer) [inline africanlii/urls.py:11]: Please use a hyphen not an underscore

**Comment by longhotsummer** [comment]
Screenshots or a video please!

File: africanlii/urls.py (hunk 3 above)
Comment (actlikewill) [reply africanlii/urls.py]: changed

**Comment by actlikewill** [comment]
I've done all the other related listing tasks in this PR.

File: africanlii/views.py (hunk 1 above)
Comment (actlikewill) [reply africanlii/views.py]: Changed. Is get_base_queryset a django builtin method or is it our own?

Comment (longhotsummer) [reply africanlii/views.py]: Our own, in FilteredDocumentListView. The base queryset is what we do faceting and counts from, before filters are applied. `get_queryset()` is the filtered base queryset.

File: africanlii/urls.py [hunk 4]
```diff
@@ -7,5 +7,14 @@
     path(
         "legislation/", views.AGPLegislationListView.as_view(), name="legislation_list"
```
Comment (longhotsummer) [inline africanlii/urls.py:8]: This can now fall away completely

**Review by longhotsummer — DISMISSED** [review]
Looks good.

I'm pretty sure /legislation can now fall away, including the view class and templates.

**Pull request merged**